apis_configs/yaml_merge.py

```diff
@@ -0,0 +1,28 @@
+'''
+Little helper that merges a couple input yaml files,
+and spits out the output to stdout.
+Gets inlined into fence-deploy.yaml to merge
+public and private config files.
+
+Use: python yaml_merge.py a.yaml b.yaml
+
+Overrides b.yaml keys with values from a.yaml
+pip install -u json
+pip install -u yaml
+
+See gen3/test/yamlMergeTest.sh - gen3 testsuite --filter yamlMerge
+
+'''
+
+import sys
+import json
+from yaml import safe_load as yaml_load
+
+config1 = yaml_load(open(sys.argv[1]))
+config2 = yaml_load(open(sys.argv[2]))
+
+if config1 != None:
+  for key in config1.keys():
```

Comment:
this will only replace top-level keys, so you can't have "secret" nested keys. there are a lot of instances in the current fence config where nested vars are a mix of public and secret. this is not ideal, but fine as an mvp. The default config handling has the same limitation, if the top-level key is there it assumes the nested vars are too :man_shrugging:

Reply:
yeah - I think it gets confusing if we start dong deep merge.  Hopefully over time we can restructure the fence-config, so that we have distinct public and private keys at the top level of the yaml file ...